### PR TITLE
[CI][Navi21] Enable Navi21 codepath for PR and Nightly CI

### PR DESCRIPTION
**Pull request by zhanglx13**
Three navi workstations are added as jenkins agents. They have the label `gfx1030w && mlir`. 
This PR only enabled navi nodes for PR and Nightly CI only.

### CODE REVIEW

**Review by krzysz00 — COMMENTED** [review]
Do you want this approved so you don't have to hound anyone for a review once it's working?

And, since we have one Navi node, how about only running Navi CI nightly and weekly?

**Comment by zhanglx13** [comment]
Yeah, please approve so that I can merge it as long as the node is fully tested.
I'll test only the navi21 codepath for nightly and weekly so that other codepaths will not be too occupied.

**Comment by krzysz00** [comment]
I more meant "don't run Navi21 tests on PRs, even after this lands, because we don't have enough nodes for that"

**Comment by zhanglx13** [comment]
@krzysz00 So now we have three tested navi21 nodes (two workstations and one server node) that are working at the PR CI stage. And since the scope of this PR is changed to PR CI only, I think it is ready to merge.

**Review by krzysz00 — APPROVED** [review]
Seems reasonable to me, approved

**Pull request merged**
